Stripe/STPShippingAddressViewController.m

```diff
@@ -366,11 +390,29 @@ - (UIView *)tableView:(__unused UITableView *)tableView viewForFooterInSection:(
 }
 
 - (CGFloat)tableView:(__unused UITableView *)tableView heightForHeaderInSection:(__unused NSInteger)section {
-    return 0.01f;
+    [self updateAddressHeaderView];
+    CGSize size = [self.addressHeaderView sizeThatFits:CGSizeMake(self.view.bounds.size.width, CGFLOAT_MAX)];
+    return size.height;
 }
 
 - (UIView *)tableView:(__unused UITableView *)tableView viewForHeaderInSection:(__unused NSInteger)section {
-    return [UIView new];
+    [self updateAddressHeaderView];
+    return self.addressHeaderView;
+}
+
+- (void)updateAddressHeaderView {
+    BOOL buttonVisible = (!self.addressViewModel.isValid && self.billingAddress != nil && !self.hasUsedBillingAddress);
```

Comment:
Possibly changing the height of the header based on billing address usage and then not always triggering a recalc of the height of the table may lead to some layout weirdness I think. You probably need a `[tableView beginUpdates]; [tableView endUpdates];` after every time the height may change or something so the table actually has consistent heights, or some other height caching method.